enhancements/dns/coredns-egressfirewall-integration.md

```diff
@@ -0,0 +1,363 @@
+``*.example.com``, then the Name of the corresponding CR will be ``wildcard.example.com``. This is done to adhere to the
+[Kubernetes object naming validations](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names).
+The ``.spec.iswildcard`` field of the CR will be set to true, even if it already exists.
+* The OVN-K master will not perform any DNS lookup for the wildcard DNS names directly.
+* The ``egressfirewall`` CoreDNS plugin will intercept the request and the response for the DNS lookups from all the pods. If
+the regular DNS name in the lookup matches with a wildcard DNS name, then the ``egressfirewall`` plugin will update the
+``.status`` of the corresponding `DNSName` CR with the regular DNS name and the corresponding current IPs along with the TTL
+and the next lookup time based on the TTL. However, this update will only take place if there is a change in the exisiting IP
+addresses or the next lookup time or both for the DNS name.
+* The OVN-K master will watch the ``DNSName`` CRs. When the ``.status`` of a ``DNSName`` CR is updated, the OVN-K master
+will update the ``AddressSet`` for the wildcard DNS name, which is linked with the ACL rule for the corresponding EgressFirewall
+rule(s).
+* The OVN-K master will also store the regular DNS name and the corresponding current IPs along with the TTL and the next time
+to lookup. Based on the next time to lookup, the OVN-K master will follow the same method as that of the regular DNS names to get the
+latest IPs and TTL information.
+
+#### Delete/Update
+
+##### Regular DNS name
+
+* An OpenShift cluster administrator deletes an EgressFirewall resource for a Namespace containing rule(s) for regular DNS
+name(s) OR updates an EgressFirewall resource for a Namespace and deletes rule(s) containing regular DNS name(s).
+* The OVN-K master will delete the ACL rule(s) corresponding to the EgressFirewall rule(s) containing the regular DNS name(s).
+* The OVN-K master will then check if the same regular DNS names are also used in the EgressFirewall rules in other Namespaces. If
+they are not used, then the OVN-K master will delete the corrresponding ``AdressSet`` for each of the DNS names in the EgressFirewall
+rules. The OVN-K master will also delete the corresponding ``DNSName`` CRs, only if the ``.spec.isregular`` field is set to true and
+the ``.spec.iswildcard`` field is set to false. If both the fields are set to true, then the CR will not be deleted and the
+``.spec.isregular`` field will be set to false.
+
+
+##### Wildcard DNS name
+
+* An OpenShift cluster administrator deletes an EgressFirewall resource for a Namespace containing rule(s) for wildcard DNS name(s)
+OR updates an EgressFirewall resource for a Namespace and deletes rule(s) containing wildcard DNS name(s).
+* The OVN-K master will delete the ACL rule(s) corresponding to the EgressFirewall rule(s) containing the wildcard DNS name(s).
```

Comment:
Are these separated somehow so that this logic will exclude subdomains that are covered both explicitly and within a wildcard?